timeline-chart/src/layer/time-graph-chart.ts

```diff
@@ -162,13 +173,62 @@ export class TimeGraphChart extends TimeGraphChartLayer {
                     this.stage.cursor = 'default';
                 }
                 if (keyBoardNavs['zoomin'].indexOf(keyPressed) >= 0) {
-                    this.adjustZoom(mousePositionX, true);
+                    let timeRange = this.adjustZoom(mousePositionX, true);
+                    if (timeRange !== undefined) {
+                        this.undoStack.push(timeRange);
+                    }
```

Comment:
I believe we could save some code repetition if we move storing the actions into the stack inside the actual actions, e.g., adjustZoom, in that case.